### PR TITLE
IC-1866 improve application insights reporting

### PR DESCRIPTION
**Pull request by tomsmyers**
## What does this pull request do?

- IC-1869 set appinsights cloud role name to 'interventions-ui'
- f46342b IC-1868 add current user id into appinsights requests
- 78ac9d4 setup the Express.Request.User type properly in passport.js
- 8a70c04 bump '@types/passport' to latest version'
- 328943e IC-1867 ignore requests for static assets in application insights
- 971e303 clean up application insights setup code
- 73f6994 update application insights sdk to 2.1.0

## What is the intent behind these changes?

improve app insights useability

### CODE REVIEW

File: server/azureAppInsights.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+function addUsernameProcessor(
+  envelope: Contracts.EnvelopeTelemetry,
+  contextObjects: { [name: string]: unknown } | undefined
+): boolean {
+  if (envelope.data.baseType === Contracts.TelemetryTypeString.Request) {
+    const username = (contextObjects?.['http.ServerRequest'] as Request)?.user?.username
+    if (username) {
+      // eslint-disable-next-line no-param-reassign
+      envelope.tags['ai.user.authUserId'] = username
```
Comment (sldblog) [inline server/azureAppInsights.ts:28]: 🔐 `username` contains our PPs' real names; could we use the user ID instead, please?

File: server/authentication/passport.ts [hunk 2]
```diff
@@ -11,6 +11,16 @@ import User from '../models/hmppsAuth/user'
 
 const authConfig = config.apis.hmppsAuth
 
+// the following code defines the Express.User type which allows @types/passport
+// to correctly augment Express.Request with the LoggedInUser created below
```
Comment (sldblog) [inline server/authentication/passport.ts:15]: Would it make sense to put this narrative in the commit message?

**Review by sldblog — COMMENTED** [review]
🎉 Nice!

One major thing about personally identifiable information (username)

File: server/azureAppInsights.ts (hunk 1 above)
Comment (tomsmyers) [reply server/azureAppInsights.ts]: done !

File: server/authentication/passport.ts (hunk 2 above)
Comment (tomsmyers) [reply server/authentication/passport.ts:15]: done!

**Review by sldblog — APPROVED** [review]

**Comment by sldblog** [comment]
(pending on build passing)

**Review by sldblog — APPROVED** [review]
image
🤣 

👍

**Comment by tomsmyers** [comment]
> image
> 🤣
> 
> 👍

@sldblog yeh gotta love npm

**Pull request merged**